### PR TITLE
Netops refactor

### PR DESCRIPTION
**Pull request by holmanb**
## Additional Context
This is a small step towards making cloud-init use reusable networking interfaces for better cross-distro abstractions and easier maintenance.

## Merge type

- [ ] Squash merge using "Proposed Commit Message"
- [x] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

**Comment by holmanb** [comment]
@cjp256 I think that you originally requested the iproute2 refactor to be multi-line for easier reading? Would you mind reviewing [this commit](https://github.com/canonical/cloud-init/pull/5117/commits/db9058f3a2f6810590b08d37c4bcc2a1259b3996) which contains that change?

File: cloudinit/net/activators.py [hunk 1]
```diff
@@ -17,15 +19,24 @@ class NoActivatorException(Exception):
     pass
 
 
-def _alter_interface(cmd, device_name) -> bool:
-    LOG.debug("Attempting command %s for device %s", cmd, device_name)
+def _alter_interface(cmd: list, device_name: str) -> bool:
+    """attempt to alter an interface using a command list"""
```
Comment (cjp256) [inline cloudinit/net/activators.py:23]: nit: case consistency

**Review by cjp256 — APPROVED** [review]
LGTM, nice change - thanks!

**Review by aciba90 — APPROVED** [review]
LGTM. Nice refactoring!

As the Merge Type is set to `rebase & merge`, would you like to merge your last commit into the previous ones?

**Comment by holmanb** [comment]
> LGTM. Nice refactoring!
> 
> As the Merge Type is set to `rebase & merge`, would you like to merge your last commit into the previous ones?

Done, thanks for the review @aciba90!

**Pull request merged**